modules/storage-filesystem/snapshot_test.go

```diff
@@ -104,6 +107,26 @@ func TestSnapshotStorage_StoreSnapshot(t *testing.T) {
 		info, err = os.Stat(expectedFilePath)
 		assert.Nil(t, err) // file exists
 		assert.Greater(t, info.Size(), int64(0))
+
+		// List all files in testDir
+		files, _ := os.ReadDir(testDir)
+
+		// Remove the files, ready for restore
+		for _, f := range files {
+			os.Remove(filepath.Join(testDir, f.Name()))
+			assert.NoFileExists(t, filepath.Join(testDir, f.Name()))
+		}
+
+		// Use the previous test snapshot to test the restore function
+		module.RestoreSnapshot(ctxSnapshot, snapshot.ID)
+
+		assert.DirExists(t, module.dataPath)
+
+		// Check that every file in the snapshot exists in testDir
+		for _, filePath := range files {
+			expectedFilePath = filepath.Join(testDir, filePath.Name())
+			assert.FileExists(t, expectedFilePath)
+		}
```

Comment:
I would make this as a separate `t.Run(...)`

Reply:
done